File: modustudy/frontend/src/features/dashboard-v2/api/reviewApi.ts
import api from '@/api/axios';

export interface OptionItem {
    id: string;
    text: string;
}

export interface QuestionDetail {
    questionNumber: number;
    questionText: string;
    questionType: 'MULTIPLE_CHOICE' | 'SHORT_ANSWER' | 'MULTIPLE_CHOICE_MULTIPLE'; // Matches backend enum
    options: OptionItem[];
    correctAnswer: string;
    explanation: string;
    category: string;
    lastReviewAt: string; // ISO date string
}

export interface ReviewItemDto {
    reviewItemId: number;
    contentType: 'COURSE_QUESTION' | 'STUDY_QUESTION';
    contentId: number;
    stability: number;
    difficulty: number;
    state: number;
    reps: number;
    lapses: number;
    nextReviewAt: string;
    question: QuestionDetail;
}

export interface TodayReviewResponse {
    items: ReviewItemDto[];
    totalCount: number;
    totalPages: number;
    number: number;
    size: number;
}

export interface ApiResponse<T> {
    status: number;
    message: string;
    data: T;
}

export const getTodayReviews = async (): Promise<TodayReviewResponse> => {
    const response = await api.get<ApiResponse<TodayReviewResponse>>('/api/v1/reviews/today');
    return response.data.data;
};

export type WrongAnswerSortType = 'MOST_WRONG' | 'FSRS_RECOMMENDED' | 'LATEST';

export const getWrongAnswers = async (
    sortType?: WrongAnswerSortType,
    page: number = 0,
    size: number = 5
): Promise<TodayReviewResponse> => {
    const params = {
        sortType,
        page,
        size
    };
    const response = await api.get<ApiResponse<TodayReviewResponse>>('/api/v1/reviews/wrong-answers', { params });
    return response.data.data;
};

export interface ReviewSubmitRequest {
    contentType: 'COURSE_QUESTION' | 'STUDY_QUESTION';
    contentId: number;
    userAnswer: string | null;
    responseTimeMs: number;
}

export interface ReviewSubmitResponse {
    reviewItemId: number;
    isCorrect: boolean;
    correctAnswer: string;
    state: number;
    stability: number;
    difficulty: number;
    scheduledDays: number;
    nextReviewAt: string;
}

export const submitReview = async (data: ReviewSubmitRequest): Promise<ReviewSubmitResponse> => {
    const response = await api.post<ApiResponse<ReviewSubmitResponse>>('/api/v1/reviews', data);
    return response.data.data;
};

export interface ReviewCourseWeaknessResponse {
    courseWeaknessStats: {
        courseId: number;
        courseName: string;
        totalReps: number;
        totalLapses: number;
    }[];
}

export const getCourseWeaknessStats = async (): Promise<ReviewCourseWeaknessResponse> => {
    const response = await api.get<ApiResponse<ReviewCourseWeaknessResponse>>('/api/v1/reviews/courses/weakness');
    return response.data.data;
};

export interface ReviewStatsResponse {
    totalItems: number;
    dueItems: number;
    newItems: number;
    learningItems: number;
    reviewItems: number;
    relearningItems: number;
    averageStability: number;
    totalReps: number;
    totalLapses: number;
    proficiency: number;
}

export const getReviewStats = async (): Promise<ReviewStatsResponse> => {
    const response = await api.get<ApiResponse<ReviewStatsResponse>>('/api/v1/reviews/stats');
    return response.data.data;
};

File: modustudy/mobile/app/src/main/java/com/ssafy/squiz/data/remote/api/ReviewApi.kt
package com.ssafy.squiz.data.remote.api

import com.ssafy.squiz.data.remote.model.*
import retrofit2.Response
import retrofit2.http.*

/**
 * 복습 (FSRS) API
 */
interface ReviewApi {

    /**
     * 오늘 복습 예정 카드 조회
     */
    @GET("api/v1/reviews/today")
    suspend fun getTodayReviews(): Response<ApiResponse<TodayReviewResponse>>

    /**
     * 틀린 문제 조회
     * @param sortType MOST_WRONG (많이 틀린 순) 또는 FSRS_RECOMMENDED (복습 우선순위)
     */
    @GET("api/v1/reviews/wrong-answers")
    suspend fun getWrongAnswers(
        @Query("sortType") sortType: String? = null
    ): Response<ApiResponse<TodayReviewResponse>>

    /**
     * 복습 결과 제출
     */
    @POST("api/v1/reviews")
    suspend fun submitReview(
        @Body request: ReviewSubmitRequest
    ): Response<ApiResponse<Unit>>

    /**
     * 복습 통계 조회
     */
    @GET("api/v1/reviews/stats")
    suspend fun getReviewStats(): Response<ApiResponse<ReviewStatsResponse>>

    /**
     * 퀴즈 목록 조회
     */
    @GET("api/v1/quizzes")
    suspend fun getQuizzes(
        @Query("studyId") studyId: Long? = null
    ): Response<ApiResponse<List<QuizDTO>>>

    /**
     * 퀴즈 상세 조회
     */
    @GET("api/v1/quizzes/{quizId}")
    suspend fun getQuizDetail(
        @Path("quizId") quizId: Long
    ): Response<ApiResponse<QuizDetailDTO>>

    /**
     * 퀴즈 결과 제출
     */
    @POST("api/v1/quizzes/{quizId}/submit")
    suspend fun submitQuizResult(
        @Path("quizId") quizId: Long,
        @Body answers: List<Int>
    ): Response<ApiResponse<QuizResultDTO>>
}
